target_decisioning_engine/context_provider.py

```diff
@@ -93,9 +104,7 @@ def create_mbox_context(mbox_request):
         return {}
 
     parameters = mbox_request.parameters or {}
-    lc_parameters = get_lower_case_attributes(parameters)
-    parameters.update(lc_parameters)
-    return parameters
+    return with_lowercase_string_values(copy(unflatten(parameters)))
```

Comment:
I don't think we need to call `copy()` here.  Because you call it inside `with_lowercase_string_values`